lib/components/joystick/joystick_directional.dart

```diff
@@ -66,77 +66,59 @@ class JoystickDirectional {
     _tileSize = size / 2;
   }
 
-  void initialize(Vector2 _screenSize, JoystickController joystickController) {
-    this._screenSize = _screenSize;
+  void initialize(Vector2 screenSize, JoystickController joystickController) {
+    _screenSize = _screenSize;
     _joystickController = joystickController;
-    final Offset osBackground =
-        Offset(margin.left, _screenSize.y - margin.bottom);
+
+    final osBackground = Offset(margin.left, _screenSize.y - margin.bottom);
     _backgroundRect = Rect.fromCircle(center: osBackground, radius: size / 2);
 
-    final Offset osKnob =
-        Offset(_backgroundRect.center.dx, _backgroundRect.center.dy);
+    final osKnob = _backgroundRect.center;
     _knobRect = Rect.fromCircle(center: osKnob, radius: size / 4);
 
     _dragPosition = _knobRect.center;
   }
 
   void render(Canvas canvas) {
-    if (_backgroundRect != null) {
-      if (_backgroundSprite != null) {
-        _backgroundSprite.renderRect(canvas, _backgroundRect);
-      } else {
-        final double radiusBackground = _backgroundRect.width / 2;
-        canvas.drawCircle(
-          Offset(_backgroundRect.left + radiusBackground,
-              _backgroundRect.top + radiusBackground),
-          radiusBackground,
-          _paintBackground,
-        );
-      }
-    }
+    JoystickUtils.renderControl(
+      canvas,
+      _backgroundSprite,
+      _backgroundRect,
+      _paintBackground,
+    );
 
-    if (_knobRect != null) {
-      if (_knobSprite != null) {
-        _knobSprite.renderRect(canvas, _knobRect);
-      } else {
-        final double radiusKnob = _knobRect.width / 2;
-        canvas.drawCircle(
-          Offset(_knobRect.left + radiusKnob, _knobRect.top + radiusKnob),
-          radiusKnob,
-          _paintKnob,
-        );
-      }
-    }
+    JoystickUtils.renderControl(
+      canvas,
+      _knobSprite,
+      _knobRect,
+      _paintKnob,
+    );
   }
 
   void update(double t) {
     if (_dragging) {
       final double _radAngle = atan2(
-          _dragPosition.dy - _backgroundRect.center.dy,
-          _dragPosition.dx - _backgroundRect.center.dx);
+        _dragPosition.dy - _backgroundRect.center.dy,
+        _dragPosition.dx - _backgroundRect.center.dx,
+      );
 
-      final double degrees = _radAngle * 180 / pi;
+      final degrees = _radAngle * 180 / pi;
```

Comment:
What is `degrees` used for, do we really need it?
Isn't it more consistent just to use radians everywhere?

Reply:
yeah it's weird. I will take a closer look to this class after